File: apps/api/schemas/agents.py
from datetime import datetime
from typing import Any
from uuid import UUID

from pydantic import BaseModel, Field


class PlatformAgentResponse(BaseModel):
    id: UUID
    agent_key: str
    name: str
    description: str | None = None
    category: str
    default_step_order: int
    is_enabled: bool
    group: str | None = None
    artifact: str | None = None
    metadata: dict[str, Any] = Field(default_factory=dict)

    model_config = {"from_attributes": True}


class PlatformAgentUpdate(BaseModel):
    is_enabled: bool | None = None


class CompanyAgentAccessItem(BaseModel):
    agent_key: str
    name: str
    category: str
    group: str | None = None
    default_step_order: int
    is_enabled: bool
    platform_enabled: bool
    assignable: bool


class CompanyAgentAccessUpdate(BaseModel):
    agent_key: str
    is_enabled: bool


class CompanyAgentsAccessUpdate(BaseModel):
    agents: list[CompanyAgentAccessUpdate]


class CompanyAgentCatalogItem(BaseModel):
    agent_key: str
    name: str
    description: str | None = None
    category: str
    group: str | None = None
    default_step_order: int
    is_granted: bool = True


class MyAgentItem(BaseModel):
    agent_key: str
    name: str
    description: str | None = None
    category: str
    group: str | None = None
    default_step_order: int
    is_assigned: bool
    is_enabled_on_platform: bool = True


class MemberAgentAccessItem(BaseModel):
    agent_key: str
    name: str
    category: str
    group: str | None = None
    default_step_order: int
    is_assigned: bool


class MemberAgentAccessUpdate(BaseModel):
    agent_key: str
    is_assigned: bool


class MemberAgentsAccessUpdate(BaseModel):
    agents: list[MemberAgentAccessUpdate]


class ProjectAgentItem(BaseModel):
    agent_key: str
    name: str
    category: str
    group: str | None = None
    step_order: int
    is_enabled: bool
    is_assigned: bool
    default_step_order: int


class ProjectAgentsUpdate(BaseModel):
    agents: list["ProjectAgentConfigItem"]


class ProjectAgentConfigItem(BaseModel):
    agent_key: str
    is_enabled: bool
    step_order: int | None = None

File: apps/web/src/types/agents.ts
export type AgentCategory = "planning" | "development" | "quality" | "delivery";

export interface PlatformAgent {
  id: string;
  agentKey: string;
  name: string;
  description: string | null;
  category: AgentCategory;
  defaultStepOrder: number;
  isEnabled: boolean;
  group: string | null;
  artifact: string | null;
}

export interface CompanyAgentAccess {
  agentKey: string;
  name: string;
  category: string;
  group: string | null;
  defaultStepOrder: number;
  isEnabled: boolean;
  platformEnabled: boolean;
  assignable: boolean;
}

export interface CompanyAgentCatalogItem {
  agentKey: string;
  name: string;
  description: string | null;
  category: string;
  group: string | null;
  defaultStepOrder: number;
  isGranted: boolean;
}

export interface MyAgentItem {
  agentKey: string;
  name: string;
  description: string | null;
  category: string;
  group: string | null;
  defaultStepOrder: number;
  isAssigned: boolean;
  isEnabledOnPlatform: boolean;
}

export interface MemberAgentAccess {
  agentKey: string;
  name: string;
  category: string;
  group: string | null;
  defaultStepOrder: number;
  isAssigned: boolean;
}

export interface ProjectAgentItem {
  agentKey: string;
  name: string;
  category: string;
  group: string | null;
  stepOrder: number;
  isEnabled: boolean;
  isAssigned: boolean;
  defaultStepOrder: number;
}

export const agentCategoryLabels: Record<AgentCategory, string> = {
  planning: "Planning",
  development: "Development",
  quality: "Quality",
  delivery: "Delivery",
};

export const agentGroupOrder = ["Planning", "Development", "Quality", "Delivery"] as const;
